splitio/client/factory.go

```diff
@@ -472,14 +472,29 @@ func setupLocalhostFactory(
 	if err != nil {
 		return nil, err
 	}
-	splitPeriod := cfg.TaskPeriods.SplitSync
 	readyChannel := make(chan int, 1)
-	splitAPI := &api.SplitAPI{SplitFetcher: local.NewFileSplitFetcher(cfg.SplitFile, logger)}
+	fileFormat := local.DefineFormat(cfg.SplitFile, logger)
+	splitAPI := &api.SplitAPI{SplitFetcher: local.NewFileSplitFetcher(cfg.SplitFile, logger, fileFormat)}
+
+	segmentStorage := &mutexmap.MMSegmentStorage{}
```

Comment:
I would replace this `segmentStorage := &mutexmap.MMSegmentStorage{}` by `segmentStorage = mutexmap.NewMMSegmentStorage()` and remove line #482

Reply:
Updated.